src/bonsaimodule.c

```diff
@@ -12,6 +12,32 @@ PyObject *LDAPEntryObj = NULL;
 PyObject *LDAPValueListObj = NULL;
 char debugmod = 0;
 
+#if !defined(WIN32) && !defined(MACOSX) && LDAP_VENDOR_VERSION > 20443
+/* The asynchronous connection build does not function properly on
+   macOS and only works on other unix systems from version 2.4.44 */
+int async_conn = 1;
+#else
+int async_conn = 0;
+#endif
+
+/* Get or set if we will attempt async connections */
+static PyObject *
+bonsai_get_async_conn(PyObject *self) {
```

Comment:
Not sure this function is necessary. Maybe I'm wrong, but I can't imagine a scenario when based on the return value of this function a program would do things differently.